workers/unity/Packages/io.improbable.gdk.core/Utility/EntitySnapshot.cs

```diff
@@ -90,7 +83,7 @@ internal EntitySnapshot(SchemaObject entityObject)
 
         internal void SerializeToSchemaObject(SchemaObject inObj)
         {
-            var handler = new SchemaEntitySerializer(this, inObj);
+            var handler = new SchemaEntitySerializer(inObj, components);
 
             foreach (var componentId in components.Keys)
```

Comment:
This would still allocate a new list no?

Reply:
Yes, but iterating over the `Dictionary` allocates the iterator anyway?

Reply:
Might used a cached one

Reply:
Dug into it a little bit, iterating over the dictionary directly should be a little bit more efficient